### DISCUSSION TITLE
Syntax Highlighting generation with antlr?

### DISCUSSION THREAD

**Discussion by sschneider-ihre-pvs** [mike-lischke/vscode-antlr4#272] [Q&A] [answered]
So this extension is for syntax highlighting for antlr grammar files. But how would I use antlr grammars for syntax highlighting of other languages or maybe any language? There once seem to be an attempt with uni-vscode but all articles I found was for writing extensions for individual languages.
Is there a way to swap the vscode TextMate code highlighting and identification with antlr?

**Comment by mike-lischke** [comment, marked as answer]
The answer heavily depends on the IDE for which syntax highlighting is to be implemented. In fact each IDE has its own rules. VS Code for example uses a declarative approach in the standard highlighter (Monarch) that is driven by a JSON file with regex patterns. So, ANTLR or any of its grammars aren't involved at all there.

But there's also semantic highlighting on top of syntax highlighting, which can make use of an ANTLR based lexer (to know the token types). But still this is not something that can be done in a generic way, unless all lexer would return the same token types (which might be possible for some standard types like strings, comments and identifiers), but usually a language has some specialities that would not be taken into account, if semantic highlight would just use a generic lexer.

**Reply by sschneider-ihre-pvs** [reply]
do you mean Monaco? Is that a yes or a no now? :D Is there maybe some Mapper that can map from antlr to textmate for vscode then?
I was looking at https://marketplace.visualstudio.com/items?itemName=evgeniypeshkov.syntax-highlighter and was thinking of such a thing would be possible with antlr too.

**Reply by mike-lischke** [reply]
The [Monarch syntax highlighter](https://microsoft.github.io/monaco-editor/monarch.html) is a part of Monaco, yes.

The linked extension also uses parsers for the languages it supports (and provides semantic highlighting). That's what you would have to do in your case too. Whether you use ANTLR or any other parser doesn't matter. However that's a more general discussion, not related to my extension.
